modules/git/attribute/batch.go

```diff
@@ -0,0 +1,211 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package attribute
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/log"
+)
+
+// BatchChecker provides a reader for check-attribute content that can be long running
+type BatchChecker struct {
+	attributesNum int
+	repo          *git.Repository
+	stdinWriter   *os.File
+	stdOut        *nulSeparatedAttributeWriter
+	ctx           context.Context
+	cancel        context.CancelFunc
+	cmd           *git.Command
+}
+
+// NewBatchChecker creates a check attribute reader for the current repository and provided commit ID
+func NewBatchChecker(repo *git.Repository, treeish string, attributes ...string) (*BatchChecker, error) {
+	ctx, cancel := context.WithCancel(repo.Ctx)
+	if len(attributes) == 0 {
+		attributes = LinguistAttributes
+	}
```

Comment:
I do not think it should accept any "default" argument.

Let the caller clearly pass the attributes they need.

Reply:
9db67492f3